### PR TITLE
ci: Switch to FW templates for build

### PR DESCRIPTION
**Pull request by ko3n1g**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -48,69 +48,25 @@ jobs:
...
-
-  cicd-cluster-clean:
-    runs-on: self-hosted-azure-builder
-    needs: [pre-flight]
+  cicd-test-container-build:
     if: ${{ github.event.label.name == 'Run CICD' || github.event_name == 'workflow_dispatch' }}
-    steps:
-    - name: Clean server from old files
-      run: |
-        docker system prune --filter "until=24h" --filter "label=nemo.library=nemo-core" --force
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:70]: Are we using the filtered prune ? How does this affect other repo's ?

Comment (pablo-garay) [reply .github/workflows/cicd-main.yml:70]: It seems this part would be missing with the current changes as they are : "label=nemo.library=nemo-core"

File: .github/workflows/cicd-main.yml [hunk 2]
```diff
@@ -48,69 +48,25 @@ jobs:
         id: all
         run: |
           echo "main=${{ contains(fromJSON(steps.test_to_run.outputs.main), 'all') }}" | tee -a "$GITHUB_OUTPUT" 
-
-  gpu-test:
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:52]: yes, lets remove this

File: .github/workflows/cicd-main.yml (hunk 1 above)
Comment (ko3n1g) [reply .github/workflows/cicd-main.yml:70]: I added this now to the template: https://github.com/NVIDIA/NeMo-FW-CI-templates/pull/3

File: .github/workflows/cicd-main.yml [hunk 3]
```diff
@@ -48,69 +48,28 @@ jobs:
...
-    runs-on: self-hosted-azure-builder
-    needs: [pre-flight]
+  cicd-test-container-build:
     if: ${{ github.event.label.name == 'Run CICD' || github.event_name == 'workflow_dispatch' }}
-    steps:
-    - name: Clean server from old files
-      run: |
-        docker system prune --filter "until=24h" --filter "label=nemo.library=nemo-core" --force
-        
+    uses: NVIDIA/NeMo-FW-CI-templates/.github/workflows/_build_container.yml@v0.1.0
```
Comment (pablo-garay) [inline .github/workflows/cicd-main.yml:54]: I have a super small request , can you explicitly pass the "24h" . I know it's default in the templates, but i think if it's explicit developers will be aware just by looking the GH code itself (without having to guess/find out by doing the extra steps of going to the FW templates) . 

So Can you make that 24h parameter passing explicitly for this one? :)

Comment (ko3n1g) [reply .github/workflows/cicd-main.yml:54]: done

**Review by pablo-garay — APPROVED** [review]

**Pull request merged**